lib/jsonapi/plugs/content_type_negotiation.ex

```diff
@@ -0,0 +1,59 @@
+defmodule JSONAPI.ContentTypeNegotiation do
+  @moduledoc """
+  Provides content type negotiation by validating the `content-type`
+  and `accept` headers.
+
+  The proper jsonapi.org content type is
+  `application/vnd.api+json`. As per [the spec](http://jsonapi.org/format/#content-negotiation-servers)
+
+  This plug does three things:
+
+  1. Returns 415 unless the content-type header is correct.
+  2. Returns 406 unless the accept header is correct.
+  3. Registers a before send hook to set the content-type if not already set.
+  """
+
+  import JSONAPI.ErrorView
+
+  import Plug.Conn
+
+  @jsonapi "application/vnd.api+json"
+  @acceptable_types [@jsonapi, "application/*", "*/*"]
```

Comment:
I guess I don't see how this would work?

"application/*" don't seem like valid accept headers? nor does "*/*" 

Also I don't have a ton of accept header experience so I'll defer to you.